protocol.go

```diff
@@ -17,6 +17,8 @@ type Protocol struct {
 	Name string
 
 	// Code is the protocol's multicodec (a normal, non-varint number).
+	//
+	// -1 is reserved for the special "unknown" protocol.
```

Comment:
This comment seems to be inconsistent with the rest of the code. e.g.:

```
ps = append(ps, Protocol{
  Name: "unknown",
  Size: -1,
  Code: 0,
})
```

IMO, -1 for unknown protocol fells better.

Reply:
Sorry, I changed my mind multiple times half-way through. I agree.